### DISCUSSION TITLE
Is an empty string valid or invalid if Data Type is string and format is specified?

### DISCUSSION THREAD

**Discussion by ydah** [OAI/OpenAPI-Specification#3324] [Q&A] [answered]
An empty string is valid as a string, but if a format is specified, would an empty string be considered an invalid value?
I would like to check, is it valid if an empty string comes to `something_date` with the following definition? I would like to ask this question, especially since [it is not explicitly stated](https://spec.openapis.org/oas/v3.0.3#data-types).

```yaml
paths:.
  '/foo/v1/bar':: ``post:''
    post:: ``post:''
      requestBody:: content
        content: application/json:: post
          application/json:: schema
            schema: object
              type: object
              properties: properties
                something_date: string
                  type: string
                  format: date
```

**Comment by charjr** [comment, marked as answer]
Looking at the RFC definition they reference: [full-date       = date-fullyear "-" date-month "-" date-mday](https://www.rfc-editor.org/rfc/rfc3339.html#page-8)

So the definition of a "full-date" would not allow an empty string. So according to the "date" format an empty string would be invalid.

**However** it is explicitly stated that [Tools that do not recognize a specific format MAY default back to the type alone, as if the format is not specified.](https://spec.openapis.org/oas/v3.0.3#data-types).

So I think the answer is: "it depends on the tool you use" as it is the tool that decides whether `format: date` means anything.

**Reply by ydah** [reply]
Thank you for explaining it. I understand.

**Reply by handrews** [reply]
Yeah this is correct- you can't really rely on `format` to do anything at all, that's the entire problem with it.  Whether it is useful is entirely tool-dependent.  (I've been advocating for replacing `format` with a variety of reliable keywords for different use cases for years, but it's too entrenched).

**Comment by handrews** [comment]
Closing this as resolved as the answer was accepted.
